app/models/user.rb

```diff
@@ -239,15 +236,12 @@ def one_signal_player_ids
   end
 
   def add_ip(new_ip)
-    unless ip_addresses.include?(new_ip)
-      ips = [new_ip, *ip_addresses].compact.first(PAST_IPS_LIMIT)
-      update_attribute(:ip_addresses, ips)
-    end
-    ip_addresses
+    UserIpaddress.where(user: self, ip_address: new_ip).first_or_create
+    UserIpaddress.where(user: self).map(&:ip_address)
   end
 
   def alts
-    User.alts_of(self)
+    UserIpaddress.where(user: self).map(&:user)
```

Comment:
This is a really complicated way of returning `self`, and won't actually find alt accounts for the user

Alts are accounts which share an IP address with the `self`, so you need to find all IPs for the user and then find all Users which also have that IP.  Can probably be done with a subquery, something like this:

```ruby
UserIpAddress.where(ip_address: ip_addresses.select(:ip_address)).includes(:user).map(&:user)
```